ruledata.go

```diff
@@ -716,6 +688,38 @@ func SetRules(s *NetlinkConnection, content []byte) error {
 					}
 				}
 			}
+		case "file_rules":
+			vi := v.([]interface{})
+			for ruleNo := range vi {
+				rule := vi[ruleNo].(map[string]interface{})
+				path := rule["path"]
+				if path == "" {
+					log.Fatalln("Watch option needs a path!")
+				}
+				// TODO: Add key addition support
+				// key := rule["key"]
+				perms := rule["permission"]
+				log.Println("Setting File Permissions", path, perms)
+				var dd AuditRuleData
+				dd.Buf = make([]byte, 0)
+				add := AUDIT_FILTER_EXIT
+				action := AUDIT_ALWAYS
+				_audit_syscalladded = true
+
+				err := AuditSetupWatchName(&dd, path.(string))
```

Comment:
I can't call `AuditAddWatchDir` here because so the `AuditSetupWatchName` runs the path for sanity checks and then decide on the type of the `filter` to be used.
Throughout libaudit both functions have been kept separate and are used for different purposes.
Both `AuditAddWatchDir` and `AuditSetupWatchName` should be provided as an API to the user as both are available from libaudit too and we don't want to break any client logic as they can bee using any on the versions.